akka-remote/src/main/java/akka/remote/artery/AeronErrorLog.java

```diff
@@ -39,24 +36,26 @@
  */
 public class AeronErrorLog
 {
-    private final File cncFile;
     final MappedByteBuffer cncByteBuffer;
-    final DirectBuffer cncMetaDataBuffer;
-    final int cncVersion;
     final AtomicBuffer buffer;
 
+    @Deprecated
     public AeronErrorLog(File cncFile)
     {
-      this.cncFile = cncFile;
+      this(cncFile, NoLogging.getInstance());
+    }
+
+    public AeronErrorLog(File cncFile, LoggingAdapter log)
+    {
       cncByteBuffer = IoUtil.mapExistingFile(cncFile, "cnc");
-      cncMetaDataBuffer = CncFileDescriptor.createMetaDataBuffer(cncByteBuffer);
-      cncVersion = cncMetaDataBuffer.getInt(CncFileDescriptor.cncVersionOffset(0));
+      final DirectBuffer cncMetaDataBuffer = CncFileDescriptor.createMetaDataBuffer(cncByteBuffer);
+      final int cncVersion = cncMetaDataBuffer.getInt(CncFileDescriptor.cncVersionOffset(0));
       buffer = CncFileDescriptor.createErrorLogBuffer(cncByteBuffer, cncMetaDataBuffer);
 
       if (CncFileDescriptor.CNC_VERSION != cncVersion)
       {
-          IoUtil.unmap(cncByteBuffer);
-          throw new IllegalStateException("CNC version not supported: file version=" + cncVersion);
+        log.warning("Aeron CnC version mismatch: compiled version = {}, file version = {}",
```

Comment:
If this is safe, would an 'info' message be more suitable? or is this 'dangerous' enough that it warrants a 'warning'?

Reply:
If the driver is shared and the CnC version is different than the client's version it will crash the JVM or make such Akka system unusable so a warning is better.

The way Java inline the constants at compile time can also make this a false flag warning, @patriknw requested this to be a warning, I'm not bias toward any log level TBH.

Reply:
I think it's a warning, as it might not work later when parsing the log records.